### PR TITLE
chore(Animation): move to function component, styles in theme are not required

### LINKED ISSUE #2247: Animation component depends on theme styles
## Bug Report

`<Animation>` depends on styles in a theme. That means that for Animation component to work in your custom theme, you need to copy & paste the Animation styles from Teams theme 💣

```jsx
const spinner = {
  keyframe: {
    from: {
      transform: "rotate(0deg)"
    },
    to: {
      transform: "rotate(360deg)"
    }
  },
  duration: "5s",
  iterationCount: "infinite"
}

ReactDOM.render(
  <>
    <Provider theme={mergeThemes(themes.teams, { animations: { spinner } })}>
      <Animation name="spinner">
        <Box>spins</Box>
      </Animation>
    </Provider>
    <Provider theme={{ animations: { spinner } }}>
      <Animation name="spinner">
        <Box>should spin as well</Box>
      </Animation>
    </Provider>
  </>,
  document.getElementById("root")
)
```

### Version
0.43.0

### Testcase
https://codesandbox.io/s/animation-component-without-teams-5s1yj

### PR DESCRIPTION
**Pull request by layershifter**
# BREAKING CHANGES

Utility styles from the `Animation` component where moved to the component itself. No upgrade path there as there should no be any custom styles for `Animation` component as custom animations are defined in the theme.

---

Fixes #2247.

Also affect affects performance improvements as styles now be recomputed only on demand.

### CODE REVIEW

**Comment by jurokapsiar** [comment]
@layershifter is this related to https://github.com/microsoft/fluent-ui-react/issues/2247?

**Comment by layershifter** [comment]
@jurokapsiar Yes, it will fix it 👍

File: packages/react/src/components/Animation/Animation.tsx [hunk 1]
```diff
@@ -139,92 +142,158 @@ export interface AnimationProps extends StyledComponentProps {
...
+      displayName: Animation.displayName,
+      props: {
+        className,
+        styles: createAnimationStyles(animation, context.theme),
+      },
+
+      disableAnimations: context.disableAnimations,
+      renderer: context.renderer,
+      rtl: context.rtl,
+      saveDebug: _.noop,
```
Comment (layershifter) [inline packages/react/src/components/Animation/Animation.tsx:203]: Should we have `debug` for `Animation` component?

File: packages/react/src/components/Animation/Animation.tsx [hunk 2]
```diff
@@ -139,92 +142,158 @@ export interface AnimationProps extends StyledComponentProps {
...
+  const unhandledProps = getUnhandledProps(Animation.handledProps, props)
+
+  const isChildrenFunction = typeof children === 'function'
+  const child =
+    childrenExist(children) &&
+    !isChildrenFunction &&
+    (React.Children.only(children) as React.ReactElement)
+
+  const element = (
+    <>
```
Comment (mnajdova) [inline packages/react/src/components/Animation/Animation.tsx]: Remove `<>`

File: packages/react/src/components/Animation/Animation.tsx [hunk 3]
```diff
@@ -139,92 +142,158 @@ export interface AnimationProps extends StyledComponentProps {
...
+        className,
+        styles: createAnimationStyles(animation, context.theme),
+      },
+
+      disableAnimations: context.disableAnimations,
+      renderer: context.renderer,
+      rtl: context.rtl,
+      saveDebug: _.noop,
+      theme: context.theme,
+      _internal_resolvedComponentVariables: context._internal_resolvedComponentVariables,
```
Comment (mnajdova) [inline packages/react/src/components/Animation/Animation.tsx]: This is removed. Please update your branch

File: packages/react/src/components/Animation/Animation.tsx [hunk 4]
```diff
@@ -139,92 +142,158 @@ export interface AnimationProps extends StyledComponentProps {
...
+    })
+  }, [
+    className,
+    context,
```
Comment (mnajdova) [inline packages/react/src/components/Animation/Animation.tsx:208]: Are we using everything from the context? Should we maybe list the things used from the context

File: packages/react/src/components/Animation/Animation.tsx (hunk 2 above)
Comment (layershifter) [reply packages/react/src/components/Animation/Animation.tsx]: Good catch, done 👍

File: packages/react/src/components/Animation/Animation.tsx (hunk 4 above)
Comment (layershifter) [reply packages/react/src/components/Animation/Animation.tsx:208]:
```
     disableAnimations: context.disableAnimations,
      renderer: context.renderer,
      rtl: context.rtl,
      theme: context.theme,
```

Almost everything... So I don't see any benefits in destructing it...

File: packages/react/src/components/Animation/Animation.tsx (hunk 1 above)
Comment (mnajdova) [reply packages/react/src/components/Animation/Animation.tsx:203]: No 👍

**Review by mnajdova — APPROVED** [review]

**Pull request merged**